graviti/utility/user.py

```diff
@@ -355,3 +357,178 @@ def update(self, __m: Any = (), **kwargs: _V) -> None:  # pylint: disable=argume
+
+    Arguments:
+        items: The items need to be stored into the UserOrderedDict.
+
+    """
+
+    __marker = object()
+
+    def __init__(
+        self, items: Union[Iterable[Tuple[str, _V]], Mapping[str, _V], None] = None
+    ) -> None:
+        self._keys: List[str] = []
+        self._data: Dict[str, _V] = {}
+
+        if not items:
+            return
+
+        iterable = items.items() if isinstance(items, Mapping) else items
+        for key, value in iterable:
+            self._keys.append(key)
+            self._data[key] = value
+
+    def __len__(self) -> int:
+        return self._data.__len__()
+
+    def __getitem__(self, key: Union[int, str]) -> _V:
+        if isinstance(key, int):
+            key = self._keys.__getitem__(key)
+
+        return self._data.__getitem__(key)
+
+    def __setitem__(self, key: Union[int, str], value: _V) -> None:
+        if isinstance(key, int):
+            key = self._keys.__getitem__(key)
+
+        if not self._data.__contains__(key):
+            self._keys.append(key)
+
+        self._data.__setitem__(key, value)
+
+    def __delitem__(self, key: Union[int, str]) -> None:
+        if isinstance(key, int):
+            key = self._keys.pop(key)
+        else:
+            self._keys.remove(key)
+
+        self._data.__delitem__(key)
+
+    def __contains__(self, key: Any) -> bool:
+        return self._data.__contains__(key)
+
+    def __iter__(self) -> Iterator[str]:
+        return self._keys.__iter__()
+
+    def __eq__(self, other: Any) -> bool:
+        if not isinstance(other, self.__class__):
+            return False
+
+        return self._keys.__eq__(other._keys) and self._data.__eq__(other._data)
+
+    def keys(self) -> Tuple[str, ...]:  # type: ignore[override]
+        """Get all element keys in order.
+
+        Returns:
+            A tuple containing all elements keys.
+
+        """
+        return tuple(self._keys)
+
+    def items(self) -> Tuple[Tuple[str, _V], ...]:  # type: ignore[override]
+        """Return a tuple of the ordered (key, value) pairs in dict.
+
+        Returns:
+            The ordered (key, value) pairs in dict.
+
+        """
+        return tuple((key, self._data.__getitem__(key)) for key in self._keys)
+
+    def values(self) -> Tuple[_V, ...]:  # type: ignore[override]
+        """Return a tuple of the values in dict.
+
+        Returns:
+            The values in dict.
+
+        """
+        return tuple(self._data.__getitem__(key) for key in self._keys)
+
+    def clear(self) -> None:
+        """Remove all items from the UserOrderedDict object."""
+        self._data.clear()
+        self._keys.clear()
+
+    @overload
+    def get(self, key: str) -> Optional[_V]:  # pylint: disable=arguments-differ
+        ...
+
+    @overload
+    def get(self, key: str, default: Union[_V, _T] = ...) -> Union[_V, _T]:
+        ...
+
+    def get(self, key: str, default: Any = None) -> Any:
```

Comment:
support int key?

Reply:
We only support int in those methods that exit in `list` as well currently.